Sources/PushNotifications.swift

```diff
@@ -13,6 +13,8 @@ import Foundation
     private let persistenceStorageOperationQueue = DispatchQueue(label: Constants.DispatchQueue.persistenceStorageOperationQueue)
     private let networkService: PushNotificationsNetworkable
 
+    private var beamsTokenProvider: BeamsTokenProvider?
```

Comment:
Does this need to be optional? Could you required it at `init`?

That would remove the `UserValidationtError.beamsTokenProviderException` error case entirely, and the logic for checking that it's not `nil`.

Reply:
I've made it optional because there might be some cases where developers will use our SDK just to set interests (without using `setUser` functionality) - in that way they don't need to care about `BeamsTokenProvider`.